otx/cli/utils/multi_gpu.py

```diff
@@ -0,0 +1,240 @@
+
+    Args:
+        key (str): arguement key.
+        value (str or None): argument value.
+        after_params (bool): whether argument should be after `param` or not.
+    """
+    if key in sys.argv:
+        if value is not None:
+            sys.argv[sys.argv.index(key) + 1] = value
+    else:
+        if not after_params and "params" in sys.argv:
+            sys.argv.insert(sys.argv.index("params"), key)
+            if value is not None:
+                sys.argv.insert(sys.argv.index("params"), value)
+        else:
+            if after_params and "params" not in sys.argv:
+                sys.argv.append("params")
+            if value is not None:
+                sys.argv.extend([key, value])
+            else:
+                sys.argv.append(key)
+
+
+class MultiGPUManager:
+    """Class to manage multi GPU training.
+
+    Args:
+        train_func (Callable): model training function.
+        gpu_ids (str): GPU indices to use. Format should be Comma-separated indices.
+        multi_gpu_port (str): port for communication between multi GPU processes.
+    """
+
+    def __init__(self, train_func: Callable, gpu_ids: str, multi_gpu_port: str):
+        self._train_func = train_func
+        self._gpu_ids = get_gpu_ids(gpu_ids)
+        self._multi_gpu_port = multi_gpu_port
+        self._main_pid = os.getpid()
+        self._processes: Optional[List[mp.Process]] = None
+
+    def is_available(self) -> bool:
+        """Check multi GPU training is available.
+
+        Returns:
+            bool:
+                whether multi GPU training is available.
+        """
+        return len(self._gpu_ids) > 1
+
+    def setup_multi_gpu_train(
+        self,
+        output_path: str,
+        optimized_hyper_parameters: Optional[ConfigurableParameters] = None,
+    ):
+        """Carry out what should be done to run multi GPU training.
+
+        Args:
+            output_path (str): output path where task output are saved.
+            optimized_hyper_parameters (ConfigurableParameters or None): hyper parameters reflecting HPO result.
+        """
+        if optimized_hyper_parameters is not None:  # if HPO is executed, optimized HPs are applied to child processes
+            self._set_optimized_hp_for_child_process(optimized_hyper_parameters)
+
+        self._processes = self._spawn_multi_gpu_processes(output_path)
+
+        signal.signal(signal.SIGINT, self._terminate_signal_handler)
+        signal.signal(signal.SIGTERM, self._terminate_signal_handler)
+
+        self.initialize_multigpu_train(0, self._gpu_ids, self._multi_gpu_port)
+
+        threading.Thread(target=self._check_child_processes_alive, daemon=True).start()
+
+    def finalize(self):
+        """Join all child processes."""
+        if self._processes is not None:
+            for p in self._processes:
+                p.join()
+
+    @staticmethod
+    def initialize_multigpu_train(rank: int, gpu_ids: List[int], multi_gpu_port: str):
```

Comment:
How about `initialize`?

Reply:
I think this function name can convey meaning better.